apiml/src/main/java/org/zowe/apiml/GatewayHealthIndicator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health.Builder;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.cloud.netflix.eureka.server.event.EurekaRegistryAvailableEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.message.log.ApimlLogger;
+import org.zowe.apiml.message.yaml.YamlMessageServiceInstance;
+import org.zowe.apiml.product.constants.CoreService;
+import org.zowe.apiml.zaas.ZaasServiceAvailableEvent;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.springframework.boot.actuate.health.Status.DOWN;
+import static org.springframework.boot.actuate.health.Status.UP;
+
+/**
+ * This class contributes the apiml component health indication to the main /application/health
+ * controlled by class {@link ApimlHealthCheckHandler} in the common package.
+ *
+ * Note: Name is kept as GatewayHealthIndicator for backwards compatibility
+ */
+@Component
+@RequiredArgsConstructor
+public class GatewayHealthIndicator extends AbstractHealthIndicator {
+
+    private static final ApimlLogger apimlLog = ApimlLogger.of(GatewayHealthIndicator.class, YamlMessageServiceInstance.getInstance());
+    private final DiscoveryClient discoveryClient;
+
+    @Value("${apiml.catalog.serviceId:}")
+    private String apiCatalogServiceId;
+
+    private AtomicBoolean discoveryAvailable = new AtomicBoolean(false);
+    private AtomicBoolean zaasAvailable = new AtomicBoolean(false);
+
+    boolean startedInformationPublished = false;
+
+    @Override
+    protected void doHealthCheck(Builder builder) throws Exception {
+        var anyCatalogIsAvailable = apiCatalogServiceId != null && !apiCatalogServiceId.isEmpty();
+        var apiCatalogUp = !this.discoveryClient.getInstances(apiCatalogServiceId).isEmpty();
```

Comment:
```suggestion
        var anyCatalogIsAvailable = !StringUtils.isBlank(apiCatalogServiceId);
        var apiCatalogUp = anyCatalogIsAvailable && !this.discoveryClient.getInstances(apiCatalogServiceId).isEmpty();
```

null could throw an exception at com.netflix.discovery.DiscoveryClient#getInstancesByVipAddress(java.lang.String, boolean, java.lang.String)

Reply:
Same as the other comment, for next PR